### ISSUE TITLE
Graph viewer: node names are not escaped

### ISSUE THREAD

**Issue by nickrolfe** [github/vscode-codeql#1333]
One of the issues I mentioned in https://github.com/github/vscode-codeql/pull/705 still seems to be present.

When the value for `semmle.label` contains backslashes, they appear to be interpreted by graphviz, when I would expect them to appear as literal backslashes. Here's an example:

```ql
/**
 * @kind graph
 * @id foo
 */
query predicate nodes(string node, string attr, string val) {
  node in ["foo\\bar", "x\\ny", "p\\q\\r\\s"] and
  attr = "semmle.label" and
  val = node
}

query predicate edges(string a, string b, string attr, string val) { none() }
```

It looks like this:
<img width="611" alt="image" src="https://user-images.githubusercontent.com/158306/166699044-f2696ffc-612b-4352-ae5e-7a275d76e509.png">

I would expect to see the actual string `x\ny` rather than `x<newline>y`.

This affected me when I was using the graph viewer to visualize parsed regexes, which often contain backslashes.

### Version

VS Code: 1.66.2
CodeQL extension: 1.6.5

**Comment by aeisenberg** [comment]
Curious....does it do what you expect if you double escape? ie- `x\\ny` --> `x\\\\ny`

**Comment by nickrolfe** [comment]
Yes, if I write the QL string `"x\\\\ny"` (which, after QL's escaping, has the value `x\\ny`), then the graph gets rendered with `x\ny`. I assume what's happening is that graphviz treats `\` as an escape character, and `\\` is the way to escape a single backslash.

**Comment by aeisenberg** [comment]
Hmmm...interesting. I think what needs to happen is that strings are escaped in vscode before rendering them to the graph.
